FordLabs/PeopleMover
# Introduced Cypress-axe, and Jest-axe

## Issue

**Discovered and resolved the following axe errors:**

Jest Axe:
_Header Component_
<img width="1609" alt="Screen Shot 2021-02-03 at 11 24 40 AM" src="https://user-images.githubusercontent.com/74197928/106777409-f7eed100-6612-11eb-9f33-417f9a4baa9e.png">

Cypress Axe:
_Error Page_
<img width="1675" alt="Screen Shot 2021-02-03 at 11 26 29 AM" src="https://user-images.githubusercontent.com/74197928/106777457-0210cf80-6613-11eb-976b-8ec5c4f33930.png">

_Space Page_
<img width="1675" alt="Screen Shot 2021-02-03 at 11 27 21 AM" src="https://user-images.githubusercontent.com/74197928/106777467-04732980-6613-11eb-9e29-353d9d8efd5b.png">


_Dashboard Page_
**Couldn't test because visiting the dashboard requires authorization. Need to make a ticket to figure out how to allow cypress testing for this page.**


## What was done

_E2E_
- [x] Upgraded Cypress from `4.12.1`, to `6.4.0` (the latest version) so cypress-axe would work properly
- [x] Installed [cypress-axe](https://www.npmjs.com/package/cypress-axe) to catch page level accessibility violations
- [x] Installed [jest-axe](https://www.npmjs.com/package/jest-axe) for unit tests  to catch component level accessibility violations

_ Unit_
- [x] Wrote page level cypress-axe tests for the follow pages: Landing, Space, and Error pages
- [x] Resolved axe errors caught within those pages. 
- [x] Wrote component level jext-axe tests for the Header and Logo components
- [x] Resolved axe errors caught within those components. 

_ Functionality_
- [x] Updated the PeopleMover logo 
- so that it didn't have a role="img" but made it so that on the dashboard, it's *not* tabbable since it's not a link, but will read out "PeopleMover" if you go to it when browsing with a screen reader.
- so that on the space page, it's a link that is tabbable and the screen reader will read out "People Mover Dashboard" when a user focuses on it.
- [x] Added landmarks to the space page (main, and footer elements)
- [x] Changed the header's space name to be an h1 since a header tag is required, and ensured that it doesn't show up empty on the dashboard


## How to test
1. Run cypress `page_accessibility` tests and ensure they pass
2. Run header and peoplemover logo tests and ensure they pass
3. Run an Axe validator like the https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/ on feature branch to ensure all axe issues are resolved (minus color contrast) on the landing page, space page, and error pages specifically
3. Ensure all changes made are acceptable
